services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/attachment/AttachmentValidatorTest.kt

```diff
@@ -116,6 +122,49 @@ class AttachmentValidatorTest {
         }
     }
 
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class ValidateExtensionForType {
+        @ParameterizedTest
+        @EnumSource(ApplicationAttachmentType::class)
+        fun `accepts 'pdf' extension for all types`(attachmentType: ApplicationAttachmentType) {
+            AttachmentValidator.validateExtensionForType("file.pdf", attachmentType)
+        }
+
+        @ParameterizedTest
+        @MethodSource("failingExtensions")
+        fun `throws exception when type is other than MUU and extension is not 'pdf'`(
+            attachmentType: ApplicationAttachmentType,
+            extension: String,
+        ) {
+            assertFailure {
+                    AttachmentValidator.validateExtensionForType("file.$extension", attachmentType)
+                }
+                .all {
+                    hasClass(AttachmentInvalidException::class)
+                    messageContains("filename=file.$extension")
+                    messageContains("attachmentType=$attachmentType")
+                }
+        }
+
+        private fun failingExtensions(): List<Arguments> {
+            val extensions = listOf("jpg", "jpeg", "png", "dgn", "dwg", "docx", "txt", "gt")
+
+            return ApplicationAttachmentType.entries
+                .filterNot { it == ApplicationAttachmentType.MUU }
```

Comment:
In the description it only states the two types that must be pdfs (`LIIKENNEJARJESTELY` and `VALTAKIRJA`). Here it tests that the type is not (currently) the last one (`MUU`). Should the filter here check for those two types instead of the one other? In case there is a 4th type at some point and it can only be pdf?

Reply:
I'm betting on the potential new type having the same restrictions as `LIIKENNEJARJESTELY` and `VALTAKIRJA`. As the test is now, it tests that the new type can't have the other extensions.

Reply:
Thanks for the clarification. Yes, if adding a new type should allow other file types, this test will fail and needs to be fixed accordingly. And if it only allows pdfs, then this test will pass as it should.